hathor/p2p/node_sync.py

```diff
@@ -264,7 +265,7 @@ def stop(self) -> None:
     def is_synced(self) -> bool:
         """ Return True if we are synced.
         """
-        return self.manager.tx_storage.latest_timestamp - self.synced_timestamp <= self.sync_threshold
+        return self.manager.tx_storage.latest_timestamp - self.peer_timestamp <= self.sync_threshold
```

Comment:
Why are we changing from `synced_timestamp` to `peer_timestamp`? Even if we have a transaction/block near the peer_timestamp, we may be way behind syncing with the peer.

Reply:
> Even if we have a transaction/block near the peer_timestamp, we may be way behind syncing with the peer.

I'm not sure we can have a tx way ahead the other blocks and transactions.

As I understand, `peer_timestamp` is the peer's latest block/tx timestamp. So if we are around that timestamp, we are synced with it.

But it actually should be the absolute of `self.manager.tx_storage.latest_timestamp - self.peer_timestamp`, as this will be a negative number if we're behind the peer.

Reply:
> `is_synced` isn't supposed to be symmetrical (that is, indicate whether the other peer is in sync with us) otherwise it will get out of sync every time we get too ahead of the rest of the network (which when you're mining is not unlikely)

But we should invert it, then. Let's analyze `self.manager.tx_storage.latest_timestamp - self.peer_timestamp`. Suppose I just started my node and I'm syncing with a peer that's all synced up. `self.peer_timestamp` will be the latest timestamp, close to `datetime.now()`, while `self.manager.tx_storage.latest_timestamp` (the latest transaction I have) is smaller than that, so the difference will be negative. If it's negative, then it will be smaller than `self.sync_threshold` and the function will return `True`.

Reply:
Wait, that totally makes sense, so now I think it should not be working, it should return `True` from the start. Is `peer_timestamp` what I think it is then? I'll have to debug this better, because this shouldn't be working but it is.

Reply:
Yeah, the definitions are a bit confusing here. We should clear understand it better.

As far as I've read the code, `peer_timestamp` should be what we've discussed: the peer's latest timestamp, as reported by it.